### PR TITLE
eslint: 9.7.0 -> 9.9.1

### PR DESCRIPTION
**Pull request by onny**
## Description of changes

Update eslint and repackage using buildNpmPackage

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by onny** [comment]
Result of `nixpkgs-review pr 337888` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>documenso</li>
    <li>pingvin-share.backend</li>
    <li>postcss-cli</li>
  </ul>
</details>
<details>
  <summary>42 packages built:</summary>
  <ul>
    <li>commitlint</li>
    <li>create-cycle-app</li>
    <li>emanote (haskellPackages.emanote)</li>
    <li>emanote.data (haskellPackages.emanote.data)</li>
    <li>emanote.doc (haskellPackages.emanote.doc)</li>
    <li>eslint</li>
    <li>fast-cli</li>
    <li>haskellPackages.tailwind</li>
    <li>haskellPackages.tailwind.doc</li>
    <li>joplin</li>
    <li>lessc</li>
    <li>lighthouse</li>
    <li>lunarvim</li>
    <li>mastodon-bot</li>
    <li>mpvScripts.mpv-cheatsheet</li>
    <li>node2nix</li>
    <li>pscid</li>
    <li>pulp</li>
    <li>shout</li>
    <li>sloc</li>
    <li>styx</li>
    <li>styx.lib</li>
    <li>styx.themes</li>
    <li>theLoungePlugins.plugins.closepms</li>
    <li>theLoungePlugins.plugins.giphy</li>
    <li>theLoungePlugins.themes.flat-blue</li>
    <li>theLoungePlugins.themes.flat-dark</li>
    <li>uppy-companion</li>
    <li>vimPlugins.coc-docker</li>
    <li>vimPlugins.coc-go</li>
    <li>vimPlugins.coc-metals</li>
    <li>vimPlugins.coc-pyright</li>
    <li>vimPlugins.coc-r-lsp</li>
    <li>vimPlugins.coc-rust-analyzer</li>
    <li>vimPlugins.coc-sh</li>
    <li>vimPlugins.coc-spell-checker</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.coc-sumneko-lua</li>
    <li>vimPlugins.coc-tslint</li>
    <li>vimPlugins.coc-vetur</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
    <li>wasm-text-gen</li>
  </ul>
</details>

File: pkgs/development/node-packages/aliases.nix [hunk 1]
```diff
@@ -84,6 +84,7 @@ mapAliases {
   inherit (pkgs) dotenv-cli; # added 2024-06-26
   eask = pkgs.eask; # added 2023-08-17
   inherit (pkgs.elmPackages) elm-test;
+  eslint = pkgs.eslint; # Added 2024-08-28
   eslint_d = pkgs.eslint_d; # Added 2023-05-26
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/aliases.nix:88]:
```suggestion
  inherit (pkgs) eslint; # Added 2024-08-28
  inherit (pkgs) eslint_d; # Added 2023-05-26
```

File: pkgs/by-name/es/eslint/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    cp ${./package-lock.json} package-lock.json
+  '';
+
+  npmDepsHash = "sha256-/E1JUsbPyHzWJ4kuNRg/blYRaAdATYbk+jnJFJyzHLE=";
+
+  dontNpmBuild = true;
+  dontNpmPrune = true;
+
+  meta = {
+    description = " Find and fix problems in your JavaScript code";
```
Comment (SuperSandro2000) [inline pkgs/by-name/es/eslint/package.nix:28]:
```suggestion
    description = "Find and fix problems in your JavaScript code";
```

**Review by SuperSandro2000 — APPROVED** [review]
otherwise LGTM

**Comment by onny** [comment]
Thank you fixed 👍

**Pull request merged**